### DISCUSSION TITLE
Apply some settings in next.config.js to production build only

### DISCUSSION THREAD

**Discussion by jinsley8** [vercel/next.js#37790] [Help] [answered]
The [Remove Console](https://nextjs.org/docs/advanced-features/compiler#remove-console) transform removes `console.*` from dev browser console, not just in production build.

Is there any way to apply this transform to production only?
```
// next.config.js
module.exports = {
  compiler: {
    removeConsole: true,
  },
}
```

I used to use `babel-plugin-transform-remove-console` to do this like so...

`.babelrc`
```
{
  "presets": ["next/babel"],
  "env": {
    "production": {
      "plugins": [["transform-remove-console", { "exclude": ["error", "warn"] }]] // removes consoles.log from app for production builds
    }
  }
}
```

**Comment by leonard** [comment, marked as answer]
Hi. It's not a json config, it's a js module. So you can use any expressions in next.config.js. Like:
```
// next.config.js
module.exports = {
  compiler: {
    removeConsole: process.env.NODE_ENV === 'production',
  },
}
```

**Comment by jinsley8** [comment]
Amazing thank you
